File: src/main/java/com/example/demo/sort/BubbleSortDemo.java
package com.example.demo.sort;

public class BubbleSortDemo {
    public static void main(String[] args) {
        int[] source = {-1, 9, -3, 6, 5,-100};
        int length = source.length;
        int tmp = 0;
        for (int i = 0; i < length - 1; i++) {
            for (int j = 0; j < length - 1 - i; j++) {
                if (source[j + 1] < source[j]) {
                    tmp = source[j];
                    source[j] = source[j + 1];
                    source[j + 1] = tmp;
                }
            }
        }
        for (int i = 0; i < length; i++) {
            System.out.print(source[i] + " ");
        }
    }
}

File: src/main/java/com/example/demo/sort/BubbleSortDemo.kt
package com.example.demo.sort

fun main() {
    var source = arrayOf(-1, -9, 3, 8, -100,-1500)
    var length = source.size
    var tmp = 0
    for (i in 0 until length - 1) {
        for (j in 0 until length - 1 - i) {
            if (source[j + 1] < source[j]) {
                tmp = source[j]
                source[j] = source[j + 1]
                source[j + 1] = tmp
            }
        }
    }
    for (i in 0 until length) {
        print(source[i].toString() + " ")
    }
}